src/edu/ucsb/nceas/metacat/index/queue/FailedIndexResubmitTimerTask.java

```diff
@@ -0,0 +1,176 @@
+                     + "create index task list since " + e.getMessage());
+        }
+        
+        try {
+            List<IndexEvent> failedDeleteEvents = IndexEventDAO.getInstance()
+                                                         .get(IndexEvent.DELETE_FAILURE_TO_QUEUE);
+            reindexFailedTasks(failedDeleteEvents);
+        } catch (SQLException e) {
+            log.error("FailedIndexResubmitTimerTask.reIndexFAiledTasks - failed to get the failure "
+                    + "delete index task list since " + e.getMessage());
+        }
+        
+    }
+
+    /**
+     * Reindex the failed index tasks stored in the index_event table
+     * @param failedEvents  the failed index event list
+     */
+    private void reindexFailedTasks(List<IndexEvent> failedEvents) {
+        if(failedEvents != null) {
+            boolean firstTime = true;
+            boolean deleteEvent = false;
+            for(IndexEvent event : failedEvents) {
+                if(event != null && event.getIdentifier() != null) {
+                    String id = event.getIdentifier().getValue();
+                    if (id != null && !id.trim().equals("")) {
+                        Date now = new Date();
+                        //if the event is too old, we will ignore it.
+                        if(event.getDate() == null || (event.getDate() != null &&
+                           ((now.getTime() - event.getDate().getTime())
+                                   <= maxAgeOfFailedIndexTask))) {
+                            try {
+                                if(firstTime && event.getAction()
+                                              .compareTo(IndexEvent.DELETE_FAILURE_TO_QUEUE) == 0) {
+                                    firstTime = false;
+                                    deleteEvent = true;
+                                }
+                                if (deleteEvent) {
+                                    //this is a delete event
+                                    deleteIndex(id);
+                                    //Succeeded and remove it from the index event table
+                                    IndexEventDAO.getInstance().remove(event.getIdentifier());
+                                } else {
+                                    IndexGeneratorTimerTask.submitIndex(id);
+                                    //Succeeded and remove it from the index event table
+                                    IndexEventDAO.getInstance().remove(event.getIdentifier());
+                                }
+                            } catch (Exception e) {
+                                log.warn("FailedIndexResubmitTimerTask.reIndexFAiledTasks - failed "
+                                        + "to submit the reindex task for the pid " + id
+                                        + " since " + e.getMessage());
+                            }
+                        } else {
```

Comment:
should something else happen in this case that the task is too old? maybe marking the task in the DB as permanently failed? otherwise it will continue to get picked up and discarded by the method

Reply:
Yeah. On the line 111, we check if a task is too old. This variable of `maxAgeOfFailedIndexTask` is configuration. The default value is 20 days I believe.

Reply:
yes but then if the task is too old, a warning is logged and nothing else happens as far as I can tell. so the task will get picked up again and another warning...and that will just continue to happen I think